### ISSUE TITLE
Unable to install monitoring v1 0.2.3 `Failed to install app cluster-alerting. Error: validation failed: unable to recognize "": no matches for kind "ServiceMonitor" in version "monitoring.coreos.com/v1"`

### ISSUE THREAD

**Issue by slickwarren** [rancher/rancher#33831]
**Rancher Server Setup**
- Rancher version: master-head ()
- Installation option (Docker install/Helm Chart): helm chart
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc): rke1 
- Proxy/Cert Details:

**Information about the Cluster**
- Kubernetes version: v1.20.9
- Cluster Type (Local/Downstream): downstream
   - If downstream, what type of cluster? (Custom/Imported or specify provider for Hosted/Infrastructure Provider): Custom

**Describe the bug**

monitoring v1 (0.2.3) fails to install.  All of the monitoring apps are in a failing state in `system -> apps`

**To Reproduce**
install monitoring v1 on a downstream cluster (0.2.3)

**Result**
some of the UI has updated (i.e., grafana links are now on the cluster dashboard page). but grafana has no metrics, prometheus only has `prometheus/io-scrape` and `istio/envoy-stats` targets up. Here are the errors from each of the apps in system -> apps

```
Failed to install app cluster-alerting. Error: validation failed: unable to recognize "": no matches for kind "ServiceMonitor" in version "monitoring.coreos.com/v1"

Failed to install app monitoring-operator. Error: UPGRADE FAILED: release: "monitoring-operator.v2" already exists

Failed to install app cluster-monitoring. Error: UPGRADE FAILED: release: "cluster-monitoring.v2" already exists

```
**Expected Result**

monitoring v1 should install properly

**Screenshots**

**Additional context**
<!--Add any other context about the problem here.—>
the `prometheus-operator-monitoring-operator`, `grafana-cluster-monitoring` and `prometheus-cluster-monitoring` workloads did not show any errors.

**Comment by jiaqiluo** [comment]
I cannot reproduce the bug in rancher:master-head b39523882 single-install

monitoring v1 `0.2.3` is installed successfully in an RKE1 custom cluster 1.20.9 
monitoring v1 `0.2.3` can be edited&upgraded succfully 

monitoring v2 `100.0.0+up16.6.0` is installed successfully in another RKE1 custom cluster 1.20.9

need more information from @slickwarren

**Comment by jiaqiluo** [comment]
here is the list of CRD for monitoring v1 
```
> k api-resources | grep -i monitoring
alertmanagers                                  monitoring.coreos.com/v1               true         Alertmanager
podmonitors                                    monitoring.coreos.com/v1               true         PodMonitor
prometheuses                                   monitoring.coreos.com/v1               true         Prometheus
prometheusrules                                monitoring.coreos.com/v1               true         PrometheusRule
servicemonitors                                monitoring.coreos.com/v1               true         ServiceMonitor
thanosrulers                                   monitoring.coreos.com/v1               true         ThanosRuler
```

**Comment by jiaqiluo** [comment]
based on the offline conversation with @slickwarren, I tried the following steps but still cannot reproduce the bug.

Steps:
- add a notifier (Slack in this case)
- add a new alert group, a new alert, and use the notifier ==> this triggers the installation of rancher-alerting and monitoring-operator
- wait for the rancher-alerting to be installed successfully 
- enable cluster monitoring 0.2.3

Results:
cluster monitoring is installed successfully, no error shows in the UI as the issue describes.

**Comment by jiaqiluo** [comment]
based on the offline conversation with @slickwarren, this bug is not reproducible anymore. 
The issue is closed.

**Comment by cbron** [comment]
reproduced

**Comment by slickwarren** [comment]
There is a similar error on 2.6.0-rc6 using v0.1.5/0.3.0:
* cattle-prometheus/exporter-kubelets-cluster-monitoring/2 are down in prometheus (0.1.5)
  * these targets are NOT down on 0.3.0.  
  * on both 0.3.0 and 1.5.0, **prometheus-ip-scrape has no targets** -> I think this is the issue 
* this does not occur on 2.5.9 with 0.2.2 installed ->  all targets are up

**Comment by jiaqiluo** [comment]
can be tested in the monitoring v1 `0.3.1`

**Comment by Auston-Ivison-Suse** [comment]
Was able to validate the fix on v2.6-headcommit:e0f73129d893b2d88dd24c9515c5516222a21e33

I was also able to reproduce the issue by going installing monitoring and deleting certain repositories. 
see screenshot (1)

After deleting those CRDs listed in the (1) screenshot, I then went to reinstall the monitoring and this gave the following error in the (2) screenshot. 

The fix the @jiaqiluo created is as follows " Instead of check the APIVersions monitoring.coreos.com/v1 , my fix is to check each CRD separately: for each crd in the list, create it if it does not exist in the cluster. In such way, we allow some of them already exist in the cluster and it does not fail the whole installing process. " 


1. ![image (1).png](https://zube.io/files/rancher/56dba6a9db84ef6996324d961b79b695-image-1.png)


2. ![Screen Shot 2021-09-23 at 3.46.01 PM.png](https://zube.io/files/rancher/b072586589ba18867a239ecd9d308535-screen-shot-2021-09-23-at-3-46-01-pm.png)
